### DISCUSSION TITLE
Add support for Pickup shipping method.

### DISCUSSION THREAD

**Discussion by LeOndaz** [saleor/saleor#14434] [Ideas]
Is there any possibility of adding support of pickup?

An overview:
- A product can exist in a warehouse. after that, it goes to a store location, how can someone see the store location? There's a `Location` model here, it has a `.stocks` that's exactly the same as the relation between `Warehouse` and `Stock`
- We add Pickup as a shipping method and modify logic as needed.

This also introduces a Location feature, so in the dashboard you can enter where you locations lie, which is I think an essential features because not all products are in warehouses and there must be some kind of tracker for this.

**Comment by krzysztofwolski** [comment]
Work on that is already in progress. 
Warehouses will have additional flag "use for pickup". Feature is at refining phase so cant give any ETA

**Comment by LeOndaz** [comment]
@krzysztofwolski you should note that a flag `use_for_pickup` implies that a store is a warehouse also, which is not the case because multiple stores can depend on a single warehouse. I've created a plugin (unfortunately), it creates a `Location` model (same  interface as warehouse, could be a polymorphic model if it wasn't a plugin though) and a `LocationStock` (same interface as stock, same polymorphic model thing...) and we could use the location and the stock both to work with stocking, allocations, ..etc with the same implementation. Can you discuss with me the why the `use_for_pickup`  flag?

**Comment by krzysztofwolski** [comment]
Using warehouses gives us:
- stock management (users will be able to check if product is in the store before payment
- we are reusing existing interfaces for stock management 
- store manager will be able to configure if pulling stock from other warehouses is possible

**Comment by LeOndaz** [comment]
@krzysztofwolski 
I understand, but we have 2 stocks, you're talking about the normal stock but I'm talking about the store stock, so a pickup is from the store, this means that the item is in the store currently, my whole talk is about store locations more of a pickup, and it will allow pickup logically, The stock should be not only in a warehouse, but also a store, how do we know that item X is in store Y 
 atm? this is the question I'm trying to answer. Correct me or discuss it with me, I'm willing to help with PRs.
